### ISSUE TITLE
need a credit at OpenAI to use local version of AgentGPT?

### ISSUE THREAD

**Issue by HappyPony** [reworkd/AgentGPT#1051]
### What piece of documentation is affected?

Setup https://docs.reworkd.ai/development/setup

#### Section "Prerequisietes" 
So that users can reconcile their desire to run AgentGPT locally with their financial capabilities with respect to pricing https://openai.com/pricing#language-models and prepare a smooth Local AgentGPT installation.

### Additional Information
I suggest that is clarified and the AgentGPT instructions are added in the the reference:
To use AgentGPT locally, you need a credit with OpenAI. You need to deposit some contribution (how much?) at OpenAI.
s. also https://discord.com/channels/1094791204958392551/1127834955125366945/1129314060912046150
### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by nczempin** [comment]
I agree that it would help to clarify that to get the OpenAI API key to work, you need to set up billing, and once you've used up the initial credits, you will incur usage based costs.

I know we don't need to copy the entire terms of service in the link, and we're not going to follow every step OpenAI make in detail.

But maybe someone can come up with a succinct phrase that we can put in brackets.

just "requires credit card" is not 100 % accurate, because IIRC, you get some free credit initially, and unlike AWS you are not forced to provide a CC before even starting on the free tier.

Maybe "(requires credit card after initial free tier)"? too long?

**Comment by jasangill1** [comment]
Hello @nczempin @HappyPony Thanks for the feedback!! We currently make the assumption that user are using there free credits that OpenAI gives!! Ill look into getting clear instructions so that people new to Openai api key system will be aware !!

**Comment by jasangill1** [comment]
Hello @nczempin @HappyPony Thanks for the feedback!! We currently make the assumption that user are using there free credits that OpenAI gives!! Ill look into getting clear instructions so that people new to Openai api key system will be aware !!

**Comment by HappyPony** [comment]
> We currently make the assumption that user are using there free credits that OpenAI gives!! 

I don't have any free credits in my OpenAI profile. I just see that my credits have expired - s. screnshot. **Do you know if as OpenAI user need to register extra for API Key?** I don't remember if I registered separately for OpenAI API Key _or OpenAI gives credits automatically when you register here https://platform.openai.com/._

> Would one of you be willing to tackle this issue !!!

Please tell me @jasangill1 what to do?

![Usage-Credits-2023-07-15 07-56-06](https://github.com/reworkd/AgentGPT/assets/3226308/505aac57-e8e9-491a-8abc-eb19d7db5b96)

**Comment by nczempin** [comment]
IIRC, you get free credits as soon as you register for the API key, and they expire after a month?

**Comment by HappyPony** [comment]
Here [How to Get ChatGPT API Key Free & Use it](https://www.mlyearning.org/chat-gpt-api-key/), 03.07.2023 everything is explained in detail.

**Comment by HappyPony** [comment]
Is my assumption correct that for the development of client side AgentGPT-_features that do not require communication with OpenAI servers_, e.g. like [the ability to rename agents after run is complete](https://github.com/reworkd/AgentGPT/issues/1030) and [Personalize agents with avatars](https://github.com/reworkd/AgentGPT/issues/1050)  i**s not a paid plan, no credit with OpenAI is necessary?**

**Comment by jasangill1** [comment]
Hello @HappyPony correct you can always develop on the client side without ur API key, you can also enable `Mock_agent` in the `.env` file to be never need ur API key while developing
